lib/file-source/src/file_watcher.rs

```diff
@@ -178,47 +132,40 @@ impl FileWatcher {
 fn read_until_with_max_size<R: BufRead + ?Sized>(
```

Comment:
Other than using `memchr`, are there behavioral changes in this function?

Reply:
I don't think so. I will undo all of this for now.